### PR TITLE
Port, Vessel, Vessel At Port and Owner Crud

### PR DESCRIPTION
**Pull request by adrnl1ne**
CRUD for Port, Vessel, Vessel At Port and Owner Crud

### CODE REVIEW

File: backend/SpareHub/Domain/Models/VesselAtPort.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 public class VesselAtPort
 {
     public required string VesselId { get; init; }
-    public required string PortId { get; init; }
+    public required string PortId { get; set; }
```
Comment (FHallengreen) [inline backend/SpareHub/Domain/Models/VesselAtPort.cs:6]: Skal skiftes tilbage til init da den er immutable

File: backend/SpareHub/Persistence/MySql/PortEntity.cs [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 
-namespace Domain.MySql;
+namespace Persistence.MySql;
 
 public class PortEntity
 {
+    [Key]
     [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
-    public int Id { get; init; }
+    public required int Id { get; set; }
```
Comment (FHallengreen) [inline backend/SpareHub/Persistence/MySql/PortEntity.cs:11]: Skal også have init

File: backend/SpareHub/Persistence/MySql/VesselEntity.cs [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
+    public required int Id { get; set; }
+    
+    public int OwnerId { get; init; } // Ensure this is set correctly when creating a VesselEntity
+    
     public required string Name { get; init; }
     public string? ImoNumber { get; init; }
     public string? Flag { get; init; }
-    public required OwnerEntity Owner { get; init; }
+    
+    // Consider making Owner nullable if you want to allow creation without it
```
Comment (FHallengreen) [inline backend/SpareHub/Persistence/MySql/VesselEntity.cs:20]: Fjern lige ChatGPT comments :P

File: backend/SpareHub/Service/MySql/Vessel/VesselMySqlService.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+using Shared;
+using Shared.DTOs.Owner;
+using Shared.DTOs.Vessel;
+using Shared.Exceptions;
+
+namespace Service.MySql.Vessel;
+
+public class VesselMySqlService(VesselMySqlRepository vesselMySqlRepository, OwnerMySqlRepository ownerMySqlRepository) : IVesselService
+{
+    /*public async Task<List<VesselResponse>> GetVesselsBySearchQuery(string? searchQuery = "")
```
Comment (FHallengreen) [inline backend/SpareHub/Service/MySql/Vessel/VesselMySqlService.cs:12]: Skal implementeres igen

**Review by FHallengreen — CHANGES_REQUESTED** [review]

**Review by FHallengreen — APPROVED** [review]
Ok

**Comment by adrnl1ne** [comment]
Merge conflicts solved

**Comment by adrnl1ne** [comment]
Required attribute not added

**Pull request merged**